examples/tests/valid/optionality_test.w

```diff
@@ -0,0 +1,4 @@
+let x: num? = 4;
+assert(x? == true);
+assert(!x? == false);
+// TODO: add a `x? == false` case when we implement some way to null initialize an optional (https://github.com/winglang/wing/pull/1495)
```

Comment:
We should be able to do this:

```js
let var y: num?;
assert(!y?);
```

Reply:
We currently have no way to assign a "nothing" value to an optional. I don't really like this:
```wing
let x: num?; // I default to nothing
```
Once the spec is merged I'll align to it and add the test.